src/components/Footer.tsx

```diff
@@ -0,0 +1,52 @@
+import { Filter } from '../types/Filter';
+import { Todo } from '../types/Todo';
+
+interface FooterProps {
+  todos: Todo[],
+  filter: Filter,
+  handleFilter: (filter: Filter) => void,
+}
+
+export const Footer = ({ todos, filter, handleFilter }: FooterProps) => {
+  const hasCompleted = todos.some(todo => todo.completed);
```

Comment:
Use useMemo() here for the hasCompleted value. It helps optimize performance by remembering the result of expensive calculations.